src/pages/archives.tsx

```diff
@@ -0,0 +1,62 @@
+import { graphql, Link } from "gatsby"
+import * as React from "react"
+import Layout from "../components/layout"
+import ArchivesSort from "../components/ArchivesSort"
+
+interface ArchivesPageProps {
+  data: {
+    allMarkdownRemark: {
+      nodes: [
+        {
```

Comment:
wow i did not realize you can go straight to `nodes` rather than `edges { node { ...` ! Awesome!